modules/core/arrow-core-data/src/test/kotlin/arrow/core/SetKTest.kt

```diff
@@ -36,6 +36,9 @@ class SetKTest : UnitSpec() {
 
     testLaws(
       ShowLaws.laws(SetK.show(), EQ) { SetK.just(it) },
+      // Investigate - StackOverflowError
+      // SemigroupLaws.laws(SetK.semigroup(), setOf(1, 2, 3).k(), setOf(4, 5, 6).k(), setOf(7, 8, 9).k(), EQ),
+      // MonoidLaws.laws(SetK.monoid(), Gen.genSetK(Gen.int()), EQ),
```

Comment:
Trying to fix this, somehow it ends up in a StackOverflowError

Reply:
That means some implementation is self-recursive because of some bad import. Go to `setk.kt` where the monoid instance is defined and chase the bug!

Reply:
https://github.com/arrow-kt/arrow/blob/master/modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/setk.kt#L26

Good chance plus is defined in terms of combine there.

Reply:
Yup, you are right. Thanks for directing me to it. I will fix it